src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -719,10 +719,6 @@ def _setup_cache(self, cache_cls, max_batch_size, max_cache_len: Optional[int] =
                 "make sure to use `sdpa` in the mean time, and open an issue at https://github.com/huggingface/transformers"
             )
 
-        if max_cache_len > self.model.causal_mask.shape[-1] or self.device != self.model.causal_mask.device:
```

Comment:
This is the same fix as: https://github.com/huggingface/transformers/pull/29806 will rebase once that PR is merged